rocket.go

```diff
@@ -68,14 +81,18 @@ func (rk *Rocket) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// get response
 	handler := rk.handlers.getHandler(rs, r.Method)
+	var resp *response.Response
 	if handler != nil {
-		handler.Handle(rs, w, r)
-		return
+		resp = handler.Handle(rs, r)
+	} else {
+		body := rk.defaultHandler.Call([]reflect.Value{})[0]
+		resp = response.New(body).Status(http.StatusNotFound)
 	}
-	// 404 Page Not Found
-	w.WriteHeader(http.StatusNotFound)
-	response := rk.defaultHandler.Call([]reflect.Value{})[0]
-	fmt.Fprint(w, response)
 
+	if rk.responseHook != nil {
+		resp = rk.responseHook.Hook(resp)
+	}
+	resp.Handle(w)
```

Comment:
This why handler might not be a good name now.

Reply:
I think this place can use `resp.Dispatch(w)` instead.